### PR TITLE
Adding parser for 1.4 model manifest

### PR DESCRIPTION
**Pull request by visheshtanksale**
Working on adding unit test for v2 parser

### CODE REVIEW

File: internal/nimparser/utils/utils.go [hunk 1]
```diff
@@ -0,0 +1,49 @@
...
+// Returns:
+// - A NIMParserInterface implementation based on the schema version.
+func GetNIMParser(data []byte) nimparser.NIMParserInterface {
+	var config nimparser.NIMSchemaManifest
+	err := yaml.Unmarshal(data, &config)
+	if err != nil {
+		return nimparserv1.NIMParser{}
+	} else {
+		schemaVersion := strings.TrimSpace(config.SchemaVersion)
+		if schemaVersion == "2.0" {
```
Comment (shivamerla) [inline internal/nimparser/utils/utils.go:44]: Better to match it with any `2.x.x`

**Comment by shivamerla** [comment]
Looks great!

File: internal/nimparser/v2/nimparser.go [hunk 2]
```diff
@@ -0,0 +1,234 @@
...
+	// If a GPU product was matched and IDs (if any) also matched, return true
+	if foundGPU {
+		return true
+	}
+
+	// If no match was found in the specified GPUs, check the discovered GPUs
+	for _, productLabel := range discoveredGPUs {
+		if productLabel != "" {
+			// match for llm nim format
+			if strings.Contains(strings.ToLower(productLabel), strings.ToLower(profile.Tags["gpu"])) {
```
Comment (shivamerla) [inline internal/nimparser/v2/nimparser.go:180]: here we need to change to check for buildable profile tag, if set to true, then we need to select this profile.

File: internal/nimparser/v1/nim_parser_test.go [hunk 3]
```diff
@@ -0,0 +1 @@
+package v1
```
Comment (slu2011) [inline internal/nimparser/v1/nim_parser_test.go:1]: Can we remove this file?

File: internal/nimparser/utils/utils.go (hunk 1 above)
Comment (visheshtanksale) [reply internal/nimparser/utils/utils.go]: Fixed

File: internal/nimparser/v2/nimparser.go (hunk 2 above)
Comment (visheshtanksale) [reply internal/nimparser/v2/nimparser.go:193]: Fixed

File: internal/nimparser/v1/nim_parser_test.go (hunk 3 above)
Comment (visheshtanksale) [reply internal/nimparser/v1/nim_parser_test.go]: Thanks. Removed it

File: internal/nimparser/utils/utils.go [hunk 4]
```diff
@@ -0,0 +1,52 @@
...
+// - A NIMParserInterface implementation based on the schema version.
+func GetNIMParser(data []byte) nimparser.NIMParserInterface {
+	var config nimparser.NIMSchemaManifest
+	err := yaml.Unmarshal(data, &config)
+	if err != nil {
+		return nimparserv1.NIMParser{}
+	} else {
+		schemaVersion := strings.TrimSpace(config.SchemaVersion)
+
+		match, _ := regexp.MatchString("2\\.*\\.*", schemaVersion)
```
Comment (shivamerla) [inline internal/nimparser/utils/utils.go:46]: can we use the semver package instead and compare the major version.

File: internal/nimparser/v2/nimparser.go (hunk 2 above)
Comment (shivamerla) [reply internal/nimparser/v2/nimparser.go:193]: looks good.

**Review by shivamerla — DISMISSED** [review]
LGTM

**Review by shivamerla — DISMISSED** [review]

**Review by mikemckiernan — COMMENTED** [review]
Good-looking work.

File: bundle/manifests/apps.nvidia.com_nimcaches.yaml [hunk 5]
```diff
@@ -266,6 +266,10 @@ spec:
                       model:
                         description: Model spec for caching
                         properties:
+                          buildable:
+                            description: Buildable indicates generic model profiles
+                              that can be optimized with trtllm for any GPUs
```
[the remark this answers is not in the document]
Comment (visheshtanksale) [reply bundle/manifests/apps.nvidia.com_nimcaches.yaml]: Thanks. Fixed

File: config/crd/bases/apps.nvidia.com_nimcaches.yaml [hunk 6]
```diff
@@ -266,6 +266,10 @@ spec:
                       model:
                         description: Model spec for caching
                         properties:
+                          buildable:
+                            description: Buildable indicates generic model profiles
+                              that can be optimized with trtllm for any GPUs
```
[the remark this answers is not in the document]
Comment (visheshtanksale) [reply config/crd/bases/apps.nvidia.com_nimcaches.yaml]: Thanks. Fixed

File: deployments/helm/k8s-nim-operator/crds/apps.nvidia.com_nimcaches.yaml [hunk 7]
```diff
@@ -266,6 +266,10 @@ spec:
                       model:
                         description: Model spec for caching
                         properties:
+                          buildable:
+                            description: Buildable indicates generic model profiles
+                              that can be optimized with trtllm for any GPUs
```
[the remark this answers is not in the document]
Comment (visheshtanksale) [reply deployments/helm/k8s-nim-operator/crds/apps.nvidia.com_nimcaches.yaml]: Thanks. Fixed

**Review by shivamerla — APPROVED** [review]

**Pull request merged**